### PR TITLE
Be set up/knex tables abdi

### PR DESCRIPTION
**Pull request by thereactgirl**
# PR Description
This pull request includes all migration files needed to create the tables of the database along with minimal seed data for the tables.​
Minimal seeding is necessary for the front end to deliver any feedback or report bugs to update the database if needed before seeding more data.
​
## Type of Change
​
Types of Change:
- [x] New feature
- [ ] Refactoring / Bug fix
- [ ] Documentation update 
​
​
## Testing
​
- [ ] Yes
- [ ] No
- [x] Not yet needed
​
** testing will come with front end calling the server after endpoints are up **
​
## Completeness
Is this feature complete
- [x] Feature Complete

​
# PR Requirements
- [x] Clean and concise commenting
- [x] Clean and comprehensive documentation included
- [x] Conforms to engineering standards and checks
- [x] There are no merge conflicts
- [x] There are no warnings
- [x] Corresponding documentation elsewhere has been updated to reflect these changes
​
​
## Requester Questions
​In my opinion, I have made the  right calls but would like opinions on the following:
Please observe any columns in the migration files (especially families table) that have `not_nullable()` or any fields that you think should have `not_nullable()`. There are a lot of potential columns in the family table specifically. Are the right ones required?  Should any of these change? 

​
## Other
[Rubric](https://www.notion.so/1fc04e4fedeb429ba873b7c68d281707?v=74054da7991341c0bf970f39410c43da)  
[Template Guidelines](https://www.notion.so/Pull-Request-Template-f9264f79e1b649b9845961b5aba3eaff)
​

### CODE REVIEW

File: config/okta.js [hunk 1]
```diff
@@ -8,3 +8,7 @@ module.exports = {
     },
   },
 };
+
```
Comment (KristineYW) [inline config/okta.js]: Are these just additional empty lines?

File: data/migrations/20200625220949_create-profile.js [hunk 2]
```diff
@@ -1,15 +0,0 @@
-exports.up = (knex) => {
```
Comment (KristineYW) [inline data/migrations/20200625220949_create-profile.js:1]: Why was this file deleted?

File: data/migrations/20201209130335_logs.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-debugger, no-console */
+/* one of the main reasons we decided to connected logs to users and not families is because
```
Comment (KristineYW) [inline data/migrations/20201209130335_logs.js:2]: Good reasoning--is the code self-explanatory after you've provided this documentation?

File: data/migrations/20201209155624_families.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (KristineYW) [inline data/migrations/20201209155624_families.js:7]: Remember to leave docs for future cohorts!

File: data/migrations/20201209160611_members.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (KristineYW) [inline data/migrations/20201209160611_members.js:13]: Remember to leave docs for future cohorts!

File: data/migrations/20201209161927_notes.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (KristineYW) [inline data/migrations/20201209161927_notes.js:14]: Remember to leave docs for future cohorts!

File: data/seeds/001-users.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+exports.seed = function (knex) {
```
Comment (KristineYW) [inline data/seeds/001-users.js:1]: Remember to leave docs for future cohorts! Why do we want to delete all existing entries?

File: data/seeds/002-families.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (KristineYW) [inline data/seeds/002-families.js]: Why do we want to delete all existing entries? Do we want to remind future cohorts that this should only be done one time before putting the code into production, or do we want to put permission restrictions on who can perform this function?

File: data/seeds/003-members.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+
+exports.seed = function(knex) {
```
Comment (KristineYW) [inline data/seeds/003-members.js]: (Why does it seem like there are three of the same DELETE EXISTING ENTRIES functions? Can this be DRY-er?)

File: package-lock.json [hunk 10]
```diff
@@ -251,7 +248,6 @@
           "version": "3.2.1",
           "resolved": "https://registry.npmjs.org/ansi-styles/-/ansi-styles-3.2.1.tgz",
           "integrity": "sha512-VT0ZI6kZRdTh8YyJw3SMbYm/u+NqfsAxEpWO0Pf9sq8/e94WxxOpPKx9FR1FlyCtOVDNOQ+8ntlqFxiRc+r5qA==",
-          "dev": true,
```
Comment (KristineYW) [inline package-lock.json:254]: What do all these "dev": true deletions mean?

File: data/seeds/000-cleanup.js [hunk 11]
```diff
@@ -0,0 +1,8 @@
+const cleaner = require('knex-cleaner');
```
Comment (KristineYW) [inline data/seeds/000-cleanup.js:1]: What does this code do?

File: data/seeds/001_profiles.js [hunk 12]
```diff
@@ -1,21 +0,0 @@
-const faker = require('faker');
```
Comment (KristineYW) [inline data/seeds/001_profiles.js:1]: Why was this deleted?

**Review by KristineYW — CHANGES_REQUESTED** [review]
Since I'm not Web, I rely on documentation to understand what is going on. Don't be afraid to be verbose! Even webbies would probably appreciate more documentation rather than less.

File: config/okta.js (hunk 1 above)
Comment (thereactgirl) [reply config/okta.js]: I believe so, it looks like the return symbol. Thanks, I will fix it today.

File: data/migrations/20201209130335_logs.js (hunk 3 above)
Comment (thereactgirl) [reply data/migrations/20201209130335_logs.js:2]: I believe so yes, however, I am open to the opinions of the rest of the web team

File: data/seeds/000-cleanup.js (hunk 11 above)
Comment (thereactgirl) [reply data/seeds/000-cleanup.js:1]: @abdimohamud

File: data/migrations/20201209161927_notes.js (hunk 6 above)
Comment (thereactgirl) [reply data/migrations/20201209161927_notes.js:14]: Should I move the comments on lines 13 and 22 to the top as docs or add additional documentation?

Comment (KristineYW) [reply data/migrations/20201209161927_notes.js:14]: I think inline comments are fine if there is something necessary to explain

File: data/seeds/000-cleanup.js (hunk 11 above)
Comment (abdimohamud) [reply data/seeds/000-cleanup.js:1]: Its a helper library to clean/reset the database tables when running a new seed and just makes sure the knex migrations/locks are resloved too after.

File: data/seeds/003-members.js (hunk 9 above)
Comment (abdimohamud) [reply data/seeds/003-members.js]: we want to be able to at least in development for now to add/test/delete data in the tables we want to use for the endpoints, and its the same reason we set up separate migrations for each table as well.

File: data/migrations/20200625220949_create-profile.js (hunk 2 above)
Comment (abdimohamud) [reply data/migrations/20200625220949_create-profile.js:1]: this was the built in table/model/router included in the SPA, we wanted to make sure all migrations were appropriately named and also related to our tables in our database

**Review by abdimohamud — APPROVED** [review]
looks good to me,  next pull I'll make sure the linters are running and neccesary docs are changed are documented/commented. Server should be running locally with routes established too

File: data/seeds/003-members.js (hunk 9 above)
Comment (rleslie1015) [reply data/seeds/003-members.js:1]: Some dependencies are only used in development mode. package lock files are automated code when installing depencies. This just means a dependency won't be installed by npm install/npm ci when running in production mode

File: package-lock.json (hunk 10 above)
Comment (thereactgirl) [reply package-lock.json:254]: Some dependencies are only used in development mode. package lock files are automated code when installing depencies. This just means a dependency won't be installed by npm install/npm ci when running in production mode

File: data/seeds/001_profiles.js (hunk 12 above)
Comment (thereactgirl) [reply data/seeds/001_profiles.js:1]: This entire file was deleted. it was starter code we are not using.

File: data/seeds/001-users.js (hunk 7 above)
Comment (thereactgirl) [reply data/seeds/001-users.js:1]: Docs have been updated, this is temporary and should definitely be documented to change when we want to persist data in heroku

**Review by KristineYW — APPROVED** [review]
